internal/pkg/mqtt/sub.go

```diff
@@ -26,27 +33,69 @@ type broker struct {
 	svc    agent.Service
 	client paho.Client
 	logger logger.Logger
+	nats   *nats.Conn
 }
 
 // NewBroker returns new MQTT broker instance.
-func NewBroker(svc agent.Service, client paho.Client, log logger.Logger) MqttBroker {
+func NewBroker(svc agent.Service, client paho.Client, nats *nats.Conn, log logger.Logger) MqttBroker {
 	return &broker{
 		svc:    svc,
 		client: client,
 		logger: log,
+		nats:   nats,
 	}
 }
 
 // Subscribe subscribes to the MQTT message broker
 func (b *broker) Subscribe(topic string) error {
-	s := b.client.Subscribe(topic, 0, b.handleMsg)
+	s := b.client.Subscribe(fmt.Sprintf("%s/%s", topic, REQ_TOPIC), 0, b.handleMsg)
 	if err := s.Error(); s.Wait() && err != nil {
 		return err
 	}
 
+	if b.nats != nil {
+		n := b.client.Subscribe(fmt.Sprintf("%s/%s", topic, NATS_TOPIC), 0, b.handleNatsMsg)
+		if err := n.Error(); n.Wait() && err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// handleNatsMsg triggered when new message is received on MQTT broker
+func (b *broker) handleNatsMsg(mc paho.Client, msg paho.Message) {
+	topic := extractNatsTopic(msg.Topic(), NATS_TOPIC)
+	b.nats.Publish(topic, msg.Payload())
+}
+
+func extractNatsTopic(topic, sub string) string {
+	i := strings.LastIndex(topic, sub)
```

Comment:
Probably better validated via regexp, similar to what is done in CoAP adapter

Reply:
this is not the same

Reply:
I am not convinced.